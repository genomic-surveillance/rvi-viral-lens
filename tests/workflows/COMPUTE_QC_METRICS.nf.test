// Copyright (C) 2023 Genome Surveillance Unit/Genome Research Ltd.

nextflow_workflow {

    name "Test Workflow COMPUTE_QC_METRICS"
    script "workflows/COMPUTE_QC_METRICS.nf"
    workflow "COMPUTE_QC_METRICS"

    test("Should run without failures") {

        when {
            params {
                meta = [
                    id:"test2.9000000",
                    sample_id:"test2",
                    taxid:9000000,
                ]
                bam_file ="${projectDir}/tests/test_data/test_alignments/test2.9000000.sorted.bam"
                bai_file ="${projectDir}/tests/test_data/test_alignments/test2.9000000.sorted.bam.bai"
                ivar_variants_file ="${projectDir}/tests/test_data/test_ivar_variants_out/test2.9000000.variants.tsv"
                fasta_file ="${projectDir}/tests/test_data/test_consensuses/test2.9000000.consensus.fa"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.of(tuple(params.meta, params.bam_file, params.bai_file, params.fasta_file, params.ivar_variants_file))
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
          