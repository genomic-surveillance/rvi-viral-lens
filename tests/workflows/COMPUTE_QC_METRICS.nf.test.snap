{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2.9000000",
                            "taxid": 9000000,
                            "sample_id": "test2",
                            "mutations": 181,
                            "insertions": 1,
                            "deletions": 0,
                            "snps": 180,
                            "ti_tv_ratio": "148/32",
                            "reads_mapped": 42165,
                            "bases_mapped": 5906987,
                            "reads_mapped_in_proper_pairs": 41094,
                            "reference_length": 2368,
                            "positions_exceeding_depth": {
                                "0": 2368,
                                "5": 2358,
                                "10": 2358,
                                "15": 2358,
                                "20": 2358,
                                "25": 2358,
                                "30": 2358,
                                "35": 2358,
                                "40": 2357,
                                "45": 2357,
                                "50": 2356,
                                "55": 2356,
                                "60": 2356,
                                "65": 2356,
                                "70": 2356,
                                "75": 2356,
                                "80": 2356,
                                "85": 2356,
                                "90": 2356,
                                "95": 2356,
                                "100": 2356
                            },
                            "percent_positions_exceeding_depth_10": 99.58,
                            "percent_non_n_bases": 99.58,
                            "longest_non_n_subsequence": 2359,
                            "reads_unmapped": 17,
                            "mean_depth_per_position": 2494,
                            "consensus_length": 2369
                        },
                        "test2.9000000.sorted.bam:md5,889f09bfe09b2d0d98c51fbbfd57dd3a",
                        "test2.9000000.sorted.bam.bai:md5,3246b03eab5b39f491e9159c2d5c583b",
                        "test2.9000000.consensus.fa:md5,8fa86ad0980cad63ea403c60ac2a3fb3",
                        "test2.9000000.variants.tsv:md5,2d6012935de46ad579f812cb9366575f",
                        "test2.9000000.qc.json:md5,184412f4f113dbc7272125ca647ff343"
                    ]
                ],
                "qc_out_ch": [
                    [
                        {
                            "id": "test2.9000000",
                            "taxid": 9000000,
                            "sample_id": "test2",
                            "mutations": 181,
                            "insertions": 1,
                            "deletions": 0,
                            "snps": 180,
                            "ti_tv_ratio": "148/32",
                            "reads_mapped": 42165,
                            "bases_mapped": 5906987,
                            "reads_mapped_in_proper_pairs": 41094,
                            "reference_length": 2368,
                            "positions_exceeding_depth": {
                                "0": 2368,
                                "5": 2358,
                                "10": 2358,
                                "15": 2358,
                                "20": 2358,
                                "25": 2358,
                                "30": 2358,
                                "35": 2358,
                                "40": 2357,
                                "45": 2357,
                                "50": 2356,
                                "55": 2356,
                                "60": 2356,
                                "65": 2356,
                                "70": 2356,
                                "75": 2356,
                                "80": 2356,
                                "85": 2356,
                                "90": 2356,
                                "95": 2356,
                                "100": 2356
                            },
                            "percent_positions_exceeding_depth_10": 99.58,
                            "percent_non_n_bases": 99.58,
                            "longest_non_n_subsequence": 2359,
                            "reads_unmapped": 17,
                            "mean_depth_per_position": 2494,
                            "consensus_length": 2369
                        },
                        "test2.9000000.sorted.bam:md5,889f09bfe09b2d0d98c51fbbfd57dd3a",
                        "test2.9000000.sorted.bam.bai:md5,3246b03eab5b39f491e9159c2d5c583b",
                        "test2.9000000.consensus.fa:md5,8fa86ad0980cad63ea403c60ac2a3fb3",
                        "test2.9000000.variants.tsv:md5,2d6012935de46ad579f812cb9366575f",
                        "test2.9000000.qc.json:md5,184412f4f113dbc7272125ca647ff343"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-08-06T12:44:33.870514389"
    }
}